### DISCUSSION TITLE
Is it possible to debug NUnit tests?

### DISCUSSION THREAD

**Discussion by PapaMufflon** [mono/sdb#78] [Questions] [answered]
I tried to do so under Ubuntu with the following command:

`sdb "args debug-test.dll" "run nunit3-console.exe"`

I put a `System.Diagnostics.Debugger.Break()` inside a test, but it does not break there. NUnit runs successfully to the end instead. I surely missed something here or isn't it possible at all?

**Comment by alexrp** [comment, marked as answer]
It should work. The only way I can imagine the breakpoint is being lost is if NUnit runs the tests in a sub-process that isn't set up for debugging or something.

**Comment by PapaMufflon** [comment]
Would sdb show me if NUnit starts a sub-process? I can only see `Inferior process '26564' ('nunit3-console.exe') started`.

Btw, `sdb` breaks on a trivial hello world executable correctly.

**Comment by alexrp** [comment]
No, I'd recommend checking something like `htop` to see if any sub-processes are spawned.

**Comment by PapaMufflon** [comment]
Uuuh, yes. A nunit-agent gets spawned. *d'oh*
Passing a `--inprocess` argument for `nunit3-console.exe` solves that issue.
Thank you for the help!
